core/templates/dev/head/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts

```diff
@@ -823,6 +831,16 @@ oppia.directive('conversationSkin', [
                 return;
               }
             }
+
+            if (ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE) {
+              LearnerAnswerInfoService.askLearnerForAnswerInfo(
+                answer, interactionRulesService);
+
+              if (LearnerAnswerInfoService.canAskLearnerForAnswerInfo()) {
```

Comment:
If possible, enter the block from line 836 only if this is also true, and just have a return here.

Reply:
Actually the first part evaluates and in next if it checks whether the evaluation was result into true or false.